controller/workqueue/defaults.go

```diff
@@ -54,45 +53,26 @@ func GetPriorityFuncBuilder(cli client.Client, objectGetter func() client.Object
 
 	return func(item interface{}) int {
 		req, ok := item.(reconcile.Request)
-		if !ok {
-			return defaultWorkQueuePriority
-		}
-
-		object := objectGetter()
-		err := cli.Get(context.Background(), req.NamespacedName, object)
-		if err != nil {
-			klog.Errorf("Failed to get object: %v, error: %v", req.NamespacedName, err)
+		if !ok || req.Namespace == "" {
 			return defaultWorkQueuePriority
 		}
 
 		var priorityLableValue string
-		labels := object.GetLabels()
-		if len(labels) != 0 {
-			priorityLableValue = labels[workQueuePriorityLabel]
-		}
-
-		if priorityLableValue == "" {
-			if req.Namespace == "" {
+		namespace := &corev1.Namespace{}
+		if err := cli.Get(context.Background(), client.ObjectKey{Name: req.Namespace}, namespace); err != nil {
```

Comment:
I see that every time the priority is obtained, it will `cli.Get()` object or namespace. This will make `Add` to queue very slow and uncontrollable. The performance of using the priority queue is actually not as good as not using it.

Reply:
We can now prioritize using the label in the namespace, which is cost-effective as it primarily uses local cache, eliminating most HTTP traffic. If any performance issues arise, I will optimize it accordingly.